uturn/app/uturn.hoon

```diff
@@ -0,0 +1,75 @@
+/-  uturn
+/+  default-agent, dbug
+=,  hmac:crypto
+|%
++$  versioned-state
+    $%  state-0
+    ==
++$  state-0  [%0 server-config=(unit server:uturn) threads=(map @ta [=path])]
++$  card  card:agent:gall
+--
+%-  agent:dbug
+=|  state=state-0
+^-  agent:gall
+=<
+|_  =bowl:gall
++*  this     .
+    default   ~(. (default-agent this %|) bowl)
+    helper    ~(. +> bowl)
+::
+++  on-init
+  `this
+++  on-save
+  ^-  vase
+  !>(state)
+++  on-load  :: on-load:default
+  |=  =vase
+  ^-  (quip card _this)
+  =/  old-state  !<(versioned-state vase)
+  ?-  -.old-state
+    %0  `this(state old-state)
+  ==
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  ?+  mark  (on-poke:default mark vase)
+  %set-server-config
+    =/  new-server-config  !<(server:uturn vase)
+    `this(server-config.state (some new-server-config))
+  ==
+::
+++  on-watch
+  |=  =path
+  ^-  (quip card _this)
+  ?>  (team:title src.bowl our.bowl)
+  ?+  path  (on-watch:default path)
+  [%get-server * ~]
+    =/  server-config  (need server-config.state)
+    =/  secret  secret.server-config
+    =/  epoch  (unt:chrono:userlib now.bowl)
+    =/  user  "{<our.bowl>}-{(scow %uv (sham 2 eny.bowl))}" 
+    =/  credential  (make-credential [ttl=86.400 time=epoch secret=secret user=user url=url.server-config])
+    :_  this
+    :~  [%give %fact ~ %credential !>(credential)]
+        [%give %kick ~[path] `src.bowl]
```

Comment:
If you intend to kick just the subscription that was established just now, you can and should omit both `%kick` parameters.

```suggestion
        [%give %kick ~ ~]
```

This is similar to how `%fact` with the empty path list, as you do, gives the result only to this new subscription.